src/Trade.sol

```diff
@@ -54,9 +54,18 @@ contract Trade is ITrade {
     }
 
 	function submitOrder(Store.Order memory params, uint256 tpPrice, uint256 slPrice) external {
-
         address user = msg.sender;
-        
+
+        // check equity
+        int256 upl = getUpl(user);
+        uint256 balance = store.getBalance(user);
+        int256 equity = int256(balance) + upl;
+        uint256 lockedMargin = store.getLockedMargin(user);
+
+        if (int256(lockedMargin + params.margin) > equity) {
+            params.margin = uint256(equity - int256(lockedMargin));
```

Comment:
Thanks, but here size must also be adjusted so leverage stays consistent. @0xlock